### PR TITLE
Remote RR header links

### PR DESCRIPTION
**Pull request by tomrf1**
With this change, the RR links space in the website header is populated using a remote module import from support-dotcom-components.
We have already tested this mechanism in DCR.

- removes the header links from the twirl template, meaning a switch from server-side to client-side rendering.
- adds a switch, to toggle the remote header in frontend + DCR

### Non-supporter:
Screen Shot 2021-05-28 at 11 18 50
Screen Shot 2021-05-28 at 11 19 15
Screen Shot 2021-05-28 at 11 19 49

### Supporter:
Screen Shot 2021-05-28 at 11 21 09
Screen Shot 2021-05-28 at 11 20 53
Screen Shot 2021-05-28 at 11 20 33

### CODE REVIEW

File: common/app/conf/switches/ABTestSwitches.scala [hunk 1]
```diff
@@ -23,14 +23,4 @@ trait ABTestSwitches {
     sellByDate = new LocalDate(2021, 12, 1),
     exposeClientSide = true,
   )
-
-  Switch(
```
Comment (tomrf1) [inline common/app/conf/switches/ABTestSwitches.scala:27]: the test has ended

File: common/app/views/fragments/header.scala.html [hunk 2]
```diff
@@ -37,86 +37,7 @@
...
-                        }"
-                        href="@getReaderRevenueUrl(SupportContribute, Header)">
-                            Contribute
-                            @fragments.inlineSvg("arrow-right", "icon")
-                        </a>
-                    }
-                </div>
-            }
-
+            <div class="new-header__cta-bar" ></div>
```
Comment (tomrf1) [inline common/app/views/fragments/header.scala.html:40]: This gets populated using the remote import

File: static/src/stylesheets/layout/nav/_cta-bar.scss [hunk 3]
```diff
@@ -32,7 +16,7 @@
     }
 
     @include mq(desktop) {
-        max-width: 310px;
+        max-width: 320px;
```
Comment (tomrf1) [inline static/src/stylesheets/layout/nav/_cta-bar.scss:19]: at smallest desktop currently the copy runs onto 2 lines. This change makes it consistent with DCR

File: static/src/stylesheets/layout/nav/_cta-bar.scss [hunk 4]
```diff
@@ -1,20 +1,4 @@
 .new-header__cta-bar {
-    .hide-support-messaging &,
-    .is-recent-contributor & {
-        display: none;
```
Comment (tomrf1) [inline static/src/stylesheets/layout/nav/_cta-bar.scss:4]: css for hiding/showing is no longer required. This decision making happens in SDC

File: static/src/javascripts/projects/common/modules/commercial/contributions-service.js [hunk 5]
```diff
@@ -245,6 +266,21 @@ const getPuzzlesBanner = (payload) => {
...
+const getHeaderLinks = (payload) => {
+    const isProd = config.get('page.isProd');
+    const url = isProd ? 'https://contributions.guardianapis.com/header' : 'https://contributions.code.dev-guardianapis.com/header';
+    const json = JSON.stringify(payload);
+
+    const forcedVariant = getForcedVariant('header');
+    const queryString = forcedVariant ? `?force=${forcedVariant}` : '';
+
+    return fetchJson(`${url}${queryString}`, {
+        method: 'post',
```
Comment (mxdvl) [inline static/src/javascripts/projects/common/modules/commercial/contributions-service.js:278]: Method verbs should be uppercase.

```suggestion
        method: 'POST',
```

**Review by mxdvl — APPROVED** [review]
Looking good from a commercial perspective.

One small note on methods, note that they are technically case-sensitive, but IIUC normalisation is done for `POST` [and a few others](https://fetch.spec.whatwg.org/#methods).

**Pull request merged**